### PR TITLE
Use size_t in THManagedMapAllocator

### PR DESCRIPTION
**Pull request by cyyever**
When reviewing the source code, I found the ptrdiff_t size in THManagedMapAllocator::THManagedMapAllocator can be changed to size_t size to avoid unnecessary casts.

### CODE REVIEW

**Review by malfet — APPROVED** [review]
Looks good to me, though it would probably be good to also add a check to `MapAllocator.cpp` as `st_size` is signed value on most OSes, as far as I can tell

File: torch/csrc/StorageSharing.cpp [hunk 1]
```diff
@@ -66,9 +66,8 @@ static PyObject* THPStorage_pyNewFilenameStorage(
     PyObject* _unused,
     PyObject* args) {
   HANDLE_TH_ERRORS
-  // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
-  long long size;
-  if (!PyArg_ParseTuple(args, "L", &size)) {
+  unsigned long long size = 0;
+  if (!PyArg_ParseTuple(args, "K", &size)) {
```
Comment (malfet) [inline torch/csrc/StorageSharing.cpp:70]: Can you add a test that validates, that passing negative value here will raise an error of sorts?

Comment (cyyever) [reply torch/csrc/StorageSharing.cpp]: We can revert to use long long and check negative

**Pull request closed without merging**